assets/src/js/admin/onboarding-wizard/app/store/reducer.js

```diff
@@ -27,7 +27,7 @@ export const reducer = ( state, action ) => {
 				},
 			};
 		case 'SET_COUNTRY':
-			saveSettingWithOnboardingAPI( 'country', action.payload.country );
+			saveSettingWithOnboardingAPI( 'base_country', action.payload.country );
```

Comment:
I updated the passed setting name to avoid the complexity of converting the value on the server.